packages/cardano-services/src/Program/programs/pgBossWorker.ts

```diff
@@ -0,0 +1,75 @@
+import { CommonProgramOptions, PosgresProgramOptions, PostgresOptionDescriptions } from '../options';
+import { HttpServer } from '../../Http/HttpServer';
+import { Logger } from 'ts-log';
+import { MissingProgramOption } from '../errors/MissingProgramOption';
+import { PgBossHttpService, PgBossServiceConfig, PgBossServiceDependencies } from '../services/pgboss';
+import { PgBossQueue, isValidQueue } from '../../PgBoss/util';
+import { SrvRecord } from 'dns';
+import { createDnsResolver } from '../utils';
+import { createLogger } from 'bunyan';
+import { getConnectionConfig, getPool } from '../services/postgres';
+import { getListen } from '../../Http/util';
+
+export const PARALLEL_JOBS_DEFAULT = 10;
+export const PG_BOSS_WORKER_API_URL_DEFAULT = new URL('http://localhost:3003');
+
+export enum PgBossWorkerOptionDescriptions {
+  ParallelJobs = 'parallel jobs to run',
+  Queues = 'comma separated queue names'
+}
+
+export type PgBossWorkerArgs = CommonProgramOptions &
+  PosgresProgramOptions & { queues?: string } & Pick<PgBossServiceConfig, 'parallelJobs'>;
+
+export interface LoadPgBossWorkerDependencies {
+  dnsResolver?: (serviceName: string) => Promise<SrvRecord>;
+  logger?: Logger;
+}
+
+const pgBossWorker = 'pg-boss-worker';
+
+export interface PgBossWorkerConfig extends PgBossServiceConfig {
+  apiUrl: URL;
+}
+
+export class PgBossWorker extends HttpServer {
+  constructor(cfg: PgBossWorkerConfig, deps: PgBossServiceDependencies) {
+    const { apiUrl } = cfg;
+    const { logger } = deps;
+    const pgBossService = new PgBossHttpService(cfg, deps);
+
+    super(
+      { listen: getListen(apiUrl), name: pgBossWorker },
+      { logger, runnableDependencies: [], services: [pgBossService] }
+    );
+  }
+}
+
+export const loadPgBossWorker = async (args: PgBossWorkerArgs, deps: LoadPgBossWorkerDependencies = {}) => {
+  const logger = deps?.logger || createLogger({ level: args.loggerMinSeverity, name: pgBossWorker });
+  const dnsResolver =
+    deps?.dnsResolver ||
+    createDnsResolver(
+      { factor: args.serviceDiscoveryBackoffFactor, maxRetryTime: args.serviceDiscoveryTimeout },
+      logger
+    );
+  const db = await getPool(dnsResolver, logger, args);
+
+  const { queues } = args;
+
+  if (!queues) throw new MissingProgramOption(pgBossWorker, PgBossWorkerOptionDescriptions.Queues);
+
+  const queuesArray = queues.split(',') as PgBossQueue[];
+
+  for (const queue of queuesArray) if (!isValidQueue(queue)) throw new Error(`Unknown queue name: '${queue}'`);
+
+  if (!db)
+    throw new MissingProgramOption(pgBossWorker, [
+      PostgresOptionDescriptions.ConnectionString,
+      PostgresOptionDescriptions.ServiceDiscoveryArgs
+    ]);
+
+  const connectionConfig$ = getConnectionConfig(dnsResolver, args);
```

Comment:
I'll check better the redundancy about `connectionConfig`.

While the check on queue names is to check that the names provided by user through the command line are correct.
The target is to let the user immediately know about he tried to execute queue `poll-metadata` rather than `pool-metadata`

Reply:
Queue names should probably be validated in cli.ts validations

Reply:
2b90ae0f0 should solve this

Reply:
Any reason not to check the length and throw in `cli.ts`?

Reply:
Just to follow the same standard: all the checks as this one are in the `load${server}` functions not in `cli.ts`

Reply:
I see. I suggest to refactor it to check in `cli.ts`, as it's already validating the arguments - no need to split this responsibility. Anyway not required for this PR, we can create a separate task for this refactor.

Reply:
The code which checks for validity of the queue names is executed only when a `--queues` command line option is provided, while this check is intended to assert the option is provided.

There is the `.makeOptionMandatory()` (documented at the end of [this paragraph](https://github.com/tj/commander.js#more-configuration)), but I guess it will throw its own error, not our standard `MissingProgramOption`.

I can't see a way to apply this refactoring without breaking our standards.

Reply:
48d52c32f should solve this